--- a/src/browser.ts
+++ b/src/browser.ts
@@ -23,15 +23,8 @@
 		outer: while (!(result = await reader.read()).done) {
 			let chunk = decoder.decode(result.value, { stream: true });
 
-			idx_boundary = buffer.length;
 			buffer += chunk;
-
-			let idx_chunk = chunk.indexOf(boundary);
-			// if the chunk has a boundary, simply use it
-			!!~idx_chunk
-				? (idx_boundary += idx_chunk)
-				: // if not lets search for it in our current buffer
-				  (idx_boundary = buffer.indexOf(boundary));
+			idx_boundary = buffer.indexOf(boundary);
 
 			payloads = [];
 			while (!!~idx_boundary) {
